spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultPublisherFactory.java

```diff
@@ -72,17 +93,36 @@ public DefaultPublisherFactory(GcpProjectIdProvider projectIdProvider,
 		this.executorProvider = executorProvider;
 		this.channelProvider = channelProvider;
 		this.credentialsProvider = credentialsProvider;
+		this.retrySettings = retrySettings;
+		this.batchingSettings = batchingSettings;
+	}
+
+	public DefaultPublisherFactory(GcpProjectIdProvider projectIdProvider,
+			ExecutorProvider executorProvider,
+			TransportChannelProvider channelProvider,
+			CredentialsProvider credentialsProvider) {
+		this(projectIdProvider, executorProvider, channelProvider, credentialsProvider, null, null);
 	}
 
 	@Override
 	public Publisher getPublisher(String topic) {
 		return this.publishers.computeIfAbsent(topic, key -> {
 			try {
-				return Publisher.newBuilder(TopicName.of(this.projectId, key))
-						.setExecutorProvider(this.executorProvider)
-						.setChannelProvider(this.channelProvider)
-						.setCredentialsProvider(this.credentialsProvider)
-						.build();
+				Publisher.Builder publisherBuilder =
+						Publisher.newBuilder(TopicName.of(this.projectId, key))
+								.setExecutorProvider(this.executorProvider)
+								.setChannelProvider(this.channelProvider)
+								.setCredentialsProvider(this.credentialsProvider);
+
+				if (this.retrySettings != null) {
```

Comment:
If they are not mandatory why they are not setters?

That is really a rule we follow in many Spring projects: if property is mandatory, it must be a part of ctor args. Otherwise setter.

I don't see any reason in a builder pattern implementation here. We need that one for cases when we may go different branches downstream chain according the current selection. More over the setters way is good for XML configuration.

Reply:
I'll change this to setters because they play along well with XML.
@meltsufin wdyt?